lib/Doctrine/DBAL/Driver/PDOConnection.php

```diff
@@ -130,4 +166,49 @@ public function requiresQueryForServerVersion()
     {
         return false;
     }
+
+    /**
+     * Tracks the last insert ID at the current state.
+     *
+     * If this PDO driver is not able to fetch the last insert ID for identity columns
+     * without influencing connection state or transaction state, this is a noop method.
+     *
+     * @return void
+     */
+    public function trackLastInsertId()
+    {
+        // We need to avoid unnecessary exception generation for drivers not supporting this feature,
+        // by temporarily disabling exception mode.
+        $this->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_SILENT);
+
+        $lastInsertId = $this->fetchLastInsertId();
+
+        // Reactivate exception mode.
+        $this->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
```

Comment:
~~~We don't support changing error mode in DBAL anyway. Everything breaks down on our side if the mode is not `\PDO::ERRMODE_EXCEPTION`, so I wouldn't bother here.~~~

Scratch that - using `PDO#getAttribute()` is simple/easy here - anything against using it?